Bottom: a64fca79968cf151a095ef7f0daedb17b99e6510
Top:    c3a361d4d4d0d5b9c733af81d8ba9bde3960ae69
Author: Ben Greear <greearb@candelatech.com>
Date:   2013-07-08 15:56:15 -0700

ath9k: Another debugging patch.

Harmless, and unlikely to hit I think.  But if it did,
it might explain the tcp-collapse bug.

Signed-off-by: Ben Greear <greearb@candelatech.com>


---

diff --git a/drivers/net/wireless/ath/ath9k/recv.c b/drivers/net/wireless/ath/ath9k/recv.c
index 1ed62c2..26d2944 100644
--- a/drivers/net/wireless/ath/ath9k/recv.c
+++ b/drivers/net/wireless/ath/ath9k/recv.c
@@ -1109,6 +1109,8 @@ int ath_rx_tasklet(struct ath_softc *sc, int flush, bool hp)
 		if (sc->rx.frag) {
 			int space = skb->len - skb_tailroom(hdr_skb);
 
+			WARN_ON_ONCE(space < 0);
+
 			if (space > 0 &&
 			    pskb_expand_head(hdr_skb, 0, space, GFP_ATOMIC) < 0) {
 				dev_kfree_skb(skb);
